app/models/comment.rb

```diff
@@ -1,4 +1,13 @@
 class Comment < ApplicationRecord
-  belongs_to :user
-  belongs_to :post
+  belongs_to :user, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  after_save :update_comment_counter_post
+  def author=(user)
+    self.user_id = user.id
+  end
+
+  def update_comment_counter_post
+    post.update(comments_counter: post.comments.count)
+  end
```

Comment:
- **Optional**: Follow the same logic here as I suggested to other classes: ([Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html))